File: customer-address-jpa/src/test/java/com/izertis/example/config/RepositoriesInMemoryConfig.java
package com.izertis.example.config;

import com.izertis.example.repository.jpa.CustomerRepository;
import com.izertis.example.repository.jpa.inmemory.CustomerRepositoryInMemory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final CustomerRepository customerRepository = new CustomerRepositoryInMemory();

    @Bean
    @Primary
    public <T extends CustomerRepository> T customerRepository() {
        return (T) customerRepository;
    }

}

File: kustomer-address-jpa/src/test/kotlin/com/izertis/example/config/RepositoriesInMemoryConfig.kt
package com.izertis.example.config

import com.izertis.example.repository.jpa.inmemory.CustomerRepositoryInMemory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

  protected val customerRepository = CustomerRepositoryInMemory()

  @Bean
  @Primary
  fun customerRepository(): CustomerRepositoryInMemory {
    return customerRepository
  }
}
